packages/h5wasm/src/utils.ts

```diff
@@ -162,83 +152,86 @@ function parseAttributes(h5wAttrs: H5WasmAttributes): Attribute[] {
   });
 }
 
-export function parseDTypeFromNumericMetadata(
-  metadata: NumericMetadata,
-): NumericType {
-  const { signed, size: length, littleEndian } = metadata;
-  const size = length * 8;
-  const endianness = littleEndian ? Endianness.LE : Endianness.BE;
+export function parseDType(metadata: Metadata): DType {
+  const { type: h5tClass, size } = metadata;
 
-  if (isIntegerMetadata(metadata)) {
-    return signed ? intType(size, endianness) : uintType(size, endianness);
+  if (h5tClass === H5TClass.Integer) {
+    const { signed, littleEndian } = metadata;
+    return intOrUintType(signed, size * 8, toEndianness(littleEndian));
```

Comment:
I'm also adding a few utilities like this to keep this `parseDType` function concise (and share the logic with h5grove later). The other two are `compoundOrCplxType()` and `enumOrBoolType()`